cedar-lean/GUIDE.md

```diff
@@ -60,3 +60,12 @@ To break up implications across multiple lines:
 ## Comments
 
 Main theorems should have a docstring comment `/-- ... -/` explaining what is proven. See [here](https://leanprover-community.github.io/contribute/doc.html) for more information on the Mathlib documentation style.
+
+
+## Proof stability
+To make version upgrades easier, strive to follow these guidelines:
+
+- Use `simp only` instead of `simp`. It's okay to use `simp` to close a goal.
```

Comment:
I'm still not the biggest fan of this advice.  It creates a burden on all uses of `simp` in the codebase, even though I claim that most of them will never be broken by any version upgrade.  It also reduces readability of our proofs.  An alternate convention would be to convert to `simp only` only when a version upgrade problem is actually encountered.  However, I don't feel strongly about this and am happy to follow the `simp only` convention if everyone else agrees on it.

Reply:
Somewhat relatedly: should we encode a guideline that we prefer `rw` to `simp only` when possible?  One advantage of `rw` is that it has a clear failure state if the given rewrite is not possible.  `simp only [h]` can still succeed without using `h`, which can lead to confusion.

Reply:
It's a convention in `Mathlib`, which we roughly try to follow. Agree it doesn't look as nice, but as Kesha pointed out, maybe we should create custom tactics for commonly used sets of `simp` rules. In support of following Mathlib, the authors have more experience than us. OTOH, Mathlib is much larger so the same rules may not apply. Anecdotally, I think a lot (I'd guess more than 25%) of our broken proofs could be fixed by this.

Reply:
I'm also ok with using `rw` when possible, but I don't know of a tool that tells you when it's possible, so I'm not sure how we'd easily check this.

Reply:
I'm happy to follow `simp only` based on Mathlib's experience. 

I use `rw` when necessary, but in general, I find it too tedious and slow to use compared to `simp only`.  My workflow tends to be to use `simp` and add all the custom lemmas needed to reduce to the desired shape.  Then, I replace this `simp` with `simp only`.  It would be slower (and longer) to replace that with `rw` in general, since `simp only` will also use, e.g., commutativity of equality to match up goals and lemma statements.  With `rw`, you'd have to apply `eq_comm` in all such places manually. I end up using `rw` very selectively, when I want to rewrite a specific part of the goal into a specific shape.

Reply:
That's good enough for me, we can recommend `simp only` over `simp` and `rw`.

I remembered another convention we established a while ago: don't `open Except`